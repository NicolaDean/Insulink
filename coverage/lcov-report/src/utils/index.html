
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">51.97% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>184/354</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20.16% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>24/119</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>37/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>180/346</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="apiHelper.js"><a href="apiHelper.js.html">apiHelper.js</a></td>
	<td data-value="92.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	</tr>

<tr>
	<td class="file high" data-value="apiQuery.js"><a href="apiQuery.js.html">apiQuery.js</a></td>
	<td data-value="86.48" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 86%"></div><div class="cover-empty" style="width: 14%"></div></div>
	</td>
	<td data-value="86.48" class="pct high">86.48%</td>
	<td data-value="37" class="abs high">32/37</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="35" class="abs high">30/35</td>
	</tr>

<tr>
	<td class="file low" data-value="auth.js"><a href="auth.js.html">auth.js</a></td>
	<td data-value="5.55" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	</tr>

<tr>
	<td class="file low" data-value="chartDataFormatter.js"><a href="chartDataFormatter.js.html">chartDataFormatter.js</a></td>
	<td data-value="5.26" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	</tr>

<tr>
	<td class="file low" data-value="firebaseQuery.js"><a href="firebaseQuery.js.html">firebaseQuery.js</a></td>
	<td data-value="46.34" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 46%"></div><div class="cover-empty" style="width: 54%"></div></div>
	</td>
	<td data-value="46.34" class="pct low">46.34%</td>
	<td data-value="82" class="abs low">38/82</td>
	<td data-value="3.84" class="pct low">3.84%</td>
	<td data-value="26" class="abs low">1/26</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="15" class="abs medium">9/15</td>
	<td data-value="48.05" class="pct low">48.05%</td>
	<td data-value="77" class="abs low">37/77</td>
	</tr>

<tr>
	<td class="file low" data-value="inputChecker.js"><a href="inputChecker.js.html">inputChecker.js</a></td>
	<td data-value="20.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="36" class="abs low">0/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	</tr>

<tr>
	<td class="file medium" data-value="insulineCalculator.js"><a href="insulineCalculator.js.html">insulineCalculator.js</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="localStoreManager.js"><a href="localStoreManager.js.html">localStoreManager.js</a></td>
	<td data-value="77.77" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="63" class="abs medium">49/63</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="77.41" class="pct medium">77.41%</td>
	<td data-value="62" class="abs medium">48/62</td>
	</tr>

<tr>
	<td class="file high" data-value="testingJsons.js"><a href="testingJsons.js.html">testingJsons.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    