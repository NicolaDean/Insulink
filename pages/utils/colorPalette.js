
export default {
    primary:"",
    secondary:"",
    black:"",
    white:"",
    background:"white"
}