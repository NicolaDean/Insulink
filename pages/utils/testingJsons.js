export const ingredientList = {
    "results": [
        {
            "id": 19400,
            "name": "banana chips",
            "image": "banana-chips.jpg"
        },
        {
            "id": 93779,
            "name": "banana liqueur",
            "image": "limoncello.jpg"
        }
    ],
    "offset": 0,
    "number": 2,
    "totalResults": 13
  }
  
  export const foodDetails ={
    "id": 9266,
    "original": "pineapples",
    "originalName": "pineapples",
    "name": "pineapples",
    "nameClean": "pineapple",
    "amount": 1.0,
    "unit": "",
    "unitShort": "",
    "unitLong": "",
    "possibleUnits": [
        "piece",
        "slice",
        "fruit",
        "g",
        "oz",
        "cup",
        "serving"
    ],
    "estimatedCost": {
        "value": 299.0,
        "unit": "US Cents"
    },
    "consistency": "solid",
    "shoppingListUnits": [
        "pieces"
    ],
    "aisle": "Produce",
    "image": "pineapple.jpg",
    "meta": [],
    "nutrition": {
        "nutrients": [
            {
                "name": "Calories",
                "amount": 452.5,
                "unit": "cal",
                "percentOfDailyNeeds": 22.63
            },
            {
                "name": "Fat",
                "amount": 100.09,
                "unit": "g",
                "percentOfDailyNeeds": 1.67
            },
            {
                "name": "Saturated Fat",
                "amount": 0.08,
                "unit": "g",
                "percentOfDailyNeeds": 0.51
            },
            {
                "name": "Carbohydrates",
                "amount": 118.74,
                "unit": "g",
                "percentOfDailyNeeds": 39.58
            },
            {
                "name": "Net Carbohydrates",
                "amount": 106.07,
                "unit": "g",
                "percentOfDailyNeeds": 38.57
            },
            {
                "name": "Sugar",
                "amount": 89.14,
                "unit": "g",
                "percentOfDailyNeeds": 99.05
            },
            {
                "name": "Cholesterol",
                "amount": 0.0,
                "unit": "mg",
                "percentOfDailyNeeds": 0.0
            },
            {
                "name": "Sodium",
                "amount": 9.05,
                "unit": "mg",
                "percentOfDailyNeeds": 0.39
            },
            {
                "name": "Protein",
                "amount": 40.89,
                "unit": "g",
                "percentOfDailyNeeds": 9.77
            },
            {
                "name": "Vitamin C",
                "amount": 432.59,
                "unit": "mg",
                "percentOfDailyNeeds": 524.35
            },
            {
                "name": "Manganese",
                "amount": 8.39,
                "unit": "mg",
                "percentOfDailyNeeds": 419.47
            },
            {
                "name": "Fiber",
                "amount": 12.67,
                "unit": "g",
                "percentOfDailyNeeds": 50.68
            },
            {
                "name": "Vitamin B6",
                "amount": 1.01,
                "unit": "mg",
                "percentOfDailyNeeds": 50.68
            },
            {
                "name": "Copper",
                "amount": 1.0,
                "unit": "mg",
                "percentOfDailyNeeds": 49.78
            },
            {
                "name": "Vitamin B1",
                "amount": 0.72,
                "unit": "mg",
                "percentOfDailyNeeds": 47.66
            },
            {
                "name": "Folate",
                "amount": 162.9,
                "unit": "µg",
                "percentOfDailyNeeds": 40.73
            },
            {
                "name": "Potassium",
                "amount": 986.45,
                "unit": "mg",
                "percentOfDailyNeeds": 28.18
            },
            {
                "name": "Magnesium",
                "amount": 108.6,
                "unit": "mg",
                "percentOfDailyNeeds": 27.15
            },
            {
                "name": "Vitamin B3",
                "amount": 4.53,
                "unit": "mg",
                "percentOfDailyNeeds": 22.63
            },
            {
                "name": "Vitamin B5",
                "amount": 1.93,
                "unit": "mg",
                "percentOfDailyNeeds": 19.28
            },
            {
                "name": "Vitamin B2",
                "amount": 0.29,
                "unit": "mg",
                "percentOfDailyNeeds": 17.04
            },
            {
                "name": "Iron",
                "amount": 2.62,
                "unit": "mg",
                "percentOfDailyNeeds": 14.58
            },
            {
                "name": "Calcium",
                "amount": 117.65,
                "unit": "mg",
                "percentOfDailyNeeds": 11.77
            },
            {
                "name": "Vitamin A",
                "amount": 524.9,
                "unit": "IU",
                "percentOfDailyNeeds": 10.5
            },
            {
                "name": "Zinc",
                "amount": 1.09,
                "unit": "mg",
                "percentOfDailyNeeds": 7.24
            },
            {
                "name": "Phosphorus",
                "amount": 72.4,
                "unit": "mg",
                "percentOfDailyNeeds": 7.24
            },
            {
                "name": "Vitamin K",
                "amount": 6.34,
                "unit": "Âµg",
                "percentOfDailyNeeds": 6.03
            },
            {
                "name": "Selenium",
                "amount": 0.91,
                "unit": "Âµg",
                "percentOfDailyNeeds": 1.29
            },
            {
                "name": "Vitamin E",
                "amount": 0.18,
                "unit": "mg",
                "percentOfDailyNeeds": 1.21
            }
        ],
        "properties": [
            {
                "name": "Glycemic Index",
                "amount": 58.67,
                "unit": ""
            },
            {
                "name": "Glycemic Load",
                "amount": 62.23,
                "unit": ""
            }
        ],
        "caloricBreakdown": {
            "percentProtein": 3.88,
            "percentFat": 1.94,
            "percentCarbs": 94.18
        },
        "weightPerServing": {
            "amount": 905,
            "unit": "g"
        }
    },
    "categoryPath": [
        "tropical fruit",
        "fruit"
    ]
  }