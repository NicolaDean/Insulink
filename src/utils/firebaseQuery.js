
import firestore from '@react-native-firebase/firestore'
import { firebase } from "@react-native-firebase/firestore";
//TABLES

const userTable = "Users";
const glicemyTable = "Glicemy";
const diaryTable = "FoodDiary";

const users = firestore().collection(userTable);

//USER QUERY:-------------------------------------------------------------------

class firebaseQuery{
        /**
     * get the user actual data using email as id
     * @param {*} email email to query
     * @returns user data
     */
    getUserData = async (email) =>{
        const user = (await users.doc(email).get()).data();

    // console.log("GLIC: " + JSON.stringify(glicemy))
        return user;
    }

    registerUser = async (id,userData) =>{
        await (users.doc(id).set(userData));
    }

    udpateUser = async (userData) =>{
        //TODO UPDATE USER DATA
    }
    //------------------------------------------------------------------------------------
    //GLICEMY QUERY:----------------------------------------------------------------------

    zeroPad = (num, places) => String(num).padStart(places, '0');

    /**
     * Get the glicemy data of a specific user
     * @param {*} email email that identify user
     * @returns an array containing the glicemy data
     */
    getUserGlicemy = async (userId,date=new Date()) =>{

        let glicemy_records = {};
        date = this.glicemyDateFormatter(date);
        glicemy_records[date] = [];
        //const today = "27-02-2022";
        const res = (await (users.doc(userId).collection(glicemyTable).doc(date).get())).data();
        
        if(res == undefined) return [];

        res.data.forEach(g => {
            glicemy_records[date].push(this.changeGlicemyTimeFormat(g));
        });

        console.log("->>>>" + JSON.stringify(glicemy_records));

        return glicemy_records;
    }

    /**
     * Add a new value of glicemy to user with userId
     * @param {*} userId identifier of user
     * @param {*} glicemyData data to add
     */
    addGlicemyValue = async (userId,glicemyData) =>{

        const id = this.glicemyDateFormatter();
        const new_glicemy = await users.doc(userId).collection(glicemyTable).doc(id).get();
        
        if(new_glicemy && new_glicemy.exists)
        {
            await new_glicemy.ref.update({
                data:firebase.firestore.FieldValue.arrayUnion(glicemyData)
            })
        }else{
            await new_glicemy.ref.set({data:[glicemyData]});
        }
        

    }

    glicemyDateFormatter = (date = new Date()) =>
    {
        const today = date;
        let id = this.zeroPad(today.getDate(),2) +"-"+ this.zeroPad(today.getMonth()+1,2) +"-"+ today.getFullYear();

        return id;
    }

    getTodayGlicemy = (glicemy) =>
    {
        const today = this.glicemyDateFormatter();

        if(glicemy==null) return null;
        
        return glicemy[today];
    }

    getTodayLastGlicemy =(glicemy) =>
    {
        if(glicemy==null) {return null};
        const today = this.glicemyDateFormatter();

        const todayGlycemia= glicemy[today];
        
        if(todayGlycemia!=undefined){
            const len=Object.keys(todayGlycemia).length;
            return todayGlycemia[len-1].value;
        }
        else return 120;
    }


    changeGlicemyTimeFormat = (glicemy,realdate=false) => 
    {
        let time = {};
        if(realdate)
        {
            time = glicemy.time;
        }
        else{
            time = new Date(glicemy.time.seconds*1000);
        }
        

        let date = this.zeroPad(time.getDate(),2) +"/"+ this.zeroPad(time.getMonth(),2);
        let hours = this.zeroPad(time.getHours(),2) +":"+ this.zeroPad(time.getMinutes(),2);

        //TODO ERROR OF DATE FORMAT CHECK!
        glicemy.time = {date:date,hours:hours};
        return glicemy;
    }

    saveFoodDiary = async (userId,date,diary) =>{
        console.log("TRY FIREBASE")
        console.log("USER : " + userId + "->" + date);
        const userDiary = await users.doc(userId).collection(diaryTable).doc(date).get();
        firestore().settings({ ignoreUndefinedProperties: true }); //INGORE UNDEFINED FIELD
        const data = {
            totMacro:diary.totMacro,
            meals:diary.meals,
            activities:diary.activities
        }

        //await users.doc(userId).collection(diaryTable).doc(date).set({data:diary},{merge: true});
       if(userDiary && userDiary.exists)
        {
            console.log("exist");
            try{
                await userDiary.ref.update(data);
            }catch(e){
                console.log("ERRORE : " + e);
            }
            
        }else{
            
            await userDiary.ref.set(data)//TODO VEDERE CHE SUCCEDE QUI 
            console.log("Saved on " + userId + "-> " + date);
        }
    }

    getFoodDiary = async (userId,date) =>{
        const res = (await (users.doc(userId).collection(diaryTable).doc(date).get())).data();
        
        if(res == undefined) return [];

        return res;
    }
}

export const FirebaseQuery = new firebaseQuery(); 