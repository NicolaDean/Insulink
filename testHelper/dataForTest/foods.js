
export const dummyAppleDeletable = {"id":{"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"},"name":"apple pie","image":"https://nix-tag-images.s3.amazonaws.com/1291_highres.jpg","cal":237,"carb":34,"fat":11,"prot":1.9,"quantity":1,"unit":"piece (1/8 of 9\" dia)","identifier":1}

export const dummyApple = {"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"};
export const dummyFootDetails ={foods:[{
    "food_name": "apple pie",
    "brand_name": null,
    "serving_qty": 1,
    "serving_unit": "piece (1/8 of 9\" dia)",
    "serving_weight_grams": 248,
    "nf_calories": 60,
    "nf_total_fat": 3,
    "nf_saturated_fat": 0.65,
    "nf_cholesterol": 12.4,
    "nf_sodium": 865.52,
    "nf_total_carbohydrate": 10,
    "nf_dietary_fiber": 0.5,
    "nf_sugars": 0.67,
    "nf_protein": 4,
    "nf_potassium": 54.56,
    "nf_p": 42.16,
    "full_nutrients": [
        {
            "attr_id": 203,
            "value": 3.1496
        },
        {
            "attr_id": 204,
            "value": 2.356
        },
        {
            "attr_id": 205,
            "value": 7.316
        },
        {
            "attr_id": 207,
            "value": 2.5048
        },
        {
            "attr_id": 208,
            "value": 62
        },
        {
            "attr_id": 221,
            "value": 0
        },
        {
            "attr_id": 255,
            "value": 232.6736
        },
        {
            "attr_id": 262,
            "value": 0
        },
        {
            "attr_id": 263,
            "value": 0
        },
        {
            "attr_id": 268,
            "value": 262.88
        },
        {
            "attr_id": 269,
            "value": 0.6696
        },
        {
            "attr_id": 291,
            "value": 0.496
        },
        {
            "attr_id": 301,
            "value": 14.88
        },
        {
            "attr_id": 303,
            "value": 1.6368
        },
        {
            "attr_id": 304,
            "value": 9.92
        },
        {
            "attr_id": 305,
            "value": 42.16
        },
        {
            "attr_id": 306,
            "value": 54.56
        },
        {
            "attr_id": 307,
            "value": 865.52
        },
        {
            "attr_id": 309,
            "value": 0.3968
        },
        {
            "attr_id": 312,
            "value": 0.1587
        },
        {
            "attr_id": 315,
            "value": 0.1265
        },
        {
            "attr_id": 317,
            "value": 11.904
        },
        {
            "attr_id": 318,
            "value": 498.48
        },
        {
            "attr_id": 319,
            "value": 2.48
        },
        {
            "attr_id": 320,
            "value": 27.28
        },
        {
            "attr_id": 321,
            "value": 295.12
        },
        {
            "attr_id": 322,
            "value": 0
        },
        {
            "attr_id": 323,
            "value": 0.0744
        },
        {
            "attr_id": 324,
            "value": 0
        },
        {
            "attr_id": 328,
            "value": 0
        },
        {
            "attr_id": 334,
            "value": 0
        },
        {
            "attr_id": 337,
            "value": 0
        },
        {
            "attr_id": 338,
            "value": 9.92
        },
        {
            "attr_id": 401,
            "value": 0
        },
        {
            "attr_id": 404,
            "value": 0.1364
        },
        {
            "attr_id": 405,
            "value": 0.1116
        },
        {
            "attr_id": 406,
            "value": 1.3392
        },
        {
            "attr_id": 410,
            "value": 0.1835
        },
        {
            "attr_id": 415,
            "value": 0.0496
        },
        {
            "attr_id": 417,
            "value": 19.84
        },
        {
            "attr_id": 418,
            "value": 0.0496
        },
        {
            "attr_id": 421,
            "value": 13.64
        },
        {
            "attr_id": 430,
            "value": 0
        },
        {
            "attr_id": 431,
            "value": 14.88
        },
        {
            "attr_id": 432,
            "value": 4.96
        },
        {
            "attr_id": 435,
            "value": 29.76
        },
        {
            "attr_id": 601,
            "value": 12.4
        },
        {
            "attr_id": 605,
            "value": 0.0074
        },
        {
            "attr_id": 606,
            "value": 0.6498
        },
        {
            "attr_id": 607,
            "value": 0
        },
        {
            "attr_id": 608,
            "value": 0
        },
        {
            "attr_id": 609,
            "value": 0
        },
        {
            "attr_id": 610,
            "value": 0
        },
        {
            "attr_id": 611,
            "value": 0.0025
        },
        {
            "attr_id": 612,
            "value": 0.0124
        },
        {
            "attr_id": 613,
            "value": 0.4861
        },
        {
            "attr_id": 614,
            "value": 0.1314
        },
        {
            "attr_id": 615,
            "value": 0.0074
        },
        {
            "attr_id": 617,
            "value": 0.8432
        },
        {
            "attr_id": 618,
            "value": 0.5952
        },
        {
            "attr_id": 619,
            "value": 0.0446
        },
        {
            "attr_id": 620,
            "value": 0.0099
        },
        {
            "attr_id": 621,
            "value": 0
        },
        {
            "attr_id": 624,
            "value": 0.005
        },
        {
            "attr_id": 625,
            "value": 0.005
        },
        {
            "attr_id": 626,
            "value": 0.129
        },
        {
            "attr_id": 627,
            "value": 0
        },
        {
            "attr_id": 628,
            "value": 0.0198
        },
        {
            "attr_id": 629,
            "value": 0
        },
        {
            "attr_id": 630,
            "value": 0.0099
        },
        {
            "attr_id": 631,
            "value": 0
        },
        {
            "attr_id": 645,
            "value": 1.0366
        },
        {
            "attr_id": 646,
            "value": 0.6547
        },
        {
            "attr_id": 652,
            "value": 0.0025
        },
        {
            "attr_id": 653,
            "value": 0.005
        },
        {
            "attr_id": 654,
            "value": 0.0025
        },
        {
            "attr_id": 672,
            "value": 0.005
        },
        {
            "attr_id": 687,
            "value": 0.0198
        },
        {
            "attr_id": 689,
            "value": 0.0025
        },
        {
            "attr_id": 697,
            "value": 0.0099
        }
    ],
    "nix_brand_name": null,
    "nix_brand_id": null,
    "nix_item_name": null,
    "nix_item_id": null,
    "upc": null,
    "consumed_at": "2017-07-13T13:46:25+00:00",
    "metadata": {},
    "source": 1,
    "ndb_no": 6419,
    "tags": {
        "item": "chicken noodle soup",
        "measure": "cup",
        "quantity": "1.0",
        "tag_id": 256
    },
    "alt_measures": [
        {
            "serving_weight": 248,
            "measure": "piece (1/8 of 9\" dia)",
            "seq": 1,
            "qty": 1
        },
        {
            "serving_weight": 248,
            "measure": "cup",
            "seq": 80,
            "qty": 1
        },
        {
            "serving_weight": 586,
            "measure": "can",
            "seq": 81,
            "qty": 1
        },
        {
            "serving_weight": 496,
            "measure": "bowl (2 cups)",
            "seq": 82,
            "qty": 1
        },
        {
            "serving_weight": 5.17,
            "measure": "tsp",
            "seq": 101,
            "qty": 1
        },
        {
            "serving_weight": 15.5,
            "measure": "tbsp",
            "seq": 102,
            "qty": 1
        }
    ],
    "lat": null,
    "lng": null,
    "meal_type": 1,
    "photo": {
        "thumb": "https://d2xdmhkmkbyw75.cloudfront.net/256_thumb.jpg",
        "highres": "https://d2xdmhkmkbyw75.cloudfront.net/256_highres.jpg"
    }
  }]};

export const dummyFoodForDiary = {};

export const dummyUser = {};