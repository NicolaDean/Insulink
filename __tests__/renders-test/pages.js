import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import Food from '../../src/pages/foodSearch/food';
import { FirebaseQuery } from '../../src/utils/firebaseQuery';
import { dummyApple,dummyAppleDeletable } from '../../testHelper/dataForTest/foods';
import { foodDetails as testDetails } from '../../src/utils/testingJsons';
import FoodDetails from '../../src/pages/foodSearch/foodDetails';

const fir = ()=>{
    return {
      doc:null,
      collection: (a) =>{return "ciao"},
    };
}


jest.mock("@react-native-firebase/firestore", () => fir);
jest.mock('@react-native-async-storage/async-storage', () => 'AsyncStorage');


const reduxx = {
  useDispatch: () => mockDispatch,
  connect: (mapStateToProps, mapDispatchToProps) => (reactComponent) => ({
    mapStateToProps,
    mapDispatchToProps: (dispatch = mockDispatch, ownProps) => mapDispatchToProps(dispatch, ownProps),
    reactComponent,
    mockDispatch
  }),
  Provider: ({children}) => children,
  dispatch : mockDispatch,
  getMock: () => mockDispatch
}
const mockDispatch = jest.fn((action) => console.log(action));
jest.mock('react-redux',() => reduxx);

describe('Render our Food Related Components',()=>{
    
    beforeEach(() => {
        //Executed before each test
    });


    test('Render Food', () => {

        //Render Deletable Food
        const deletableFood = renderer.create(<Food deletable={true} data={dummyAppleDeletable}/>).toJSON();
        expect(deletableFood).toMatchSnapshot();

        //Render Normal Food
        const normalFood = renderer.create(<Food data={dummyApple}/>).toJSON();
        expect(normalFood).toMatchSnapshot();

        //Render Sport
        //TODO
      });
});//<Food data={dummyApple}/>