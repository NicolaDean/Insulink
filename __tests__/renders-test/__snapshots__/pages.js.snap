// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render our Food Related Components Render Food 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
  type="timing"
  useNativeDriver={true}
  value={0}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 3,
        "paddingVertical": 10,
      }
    }
    testID="TouchableFood"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 30,
            "elevation": 4,
            "height": 300,
            "justifyContent": "center",
            "marginHorizontal": 3,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
            "width": 300,
          },
          Object {
            "height": 300,
            "width": 300,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 13,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        apple pie
      </Text>
      <Image
        source={
          Object {
            "uri": "https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text>
        1 - piece (1/8 of 9" dia)
      </Text>
    </View>
  </View>
</View>
`;

exports[`Render our Food Related Components Render Food 2`] = `
<View
  collapsable={false}
  style={
    Object {
      "transform": Array [
        Object {
          "translateX": 0,
        },
      ],
    }
  }
  type="timing"
  useNativeDriver={true}
  value={0}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "margin": 3,
        "paddingVertical": 10,
      }
    }
    testID="TouchableFood"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 30,
            "elevation": 4,
            "height": 300,
            "justifyContent": "center",
            "marginHorizontal": 3,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.23,
            "shadowRadius": 2.62,
            "width": 300,
          },
          Object {
            "height": 225,
            "paddingTop": 20,
            "width": 225,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 13,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        apple pie
      </Text>
      <Image
        source={
          Object {
            "uri": "https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg",
          }
        }
        style={
          Object {
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text />
    </View>
  </View>
</View>
`;
